leshan-client-core/src/main/java/org/eclipse/leshan/client/object/Oscore.java

```diff
@@ -32,78 +44,140 @@
 public class Oscore extends BaseInstanceEnabler {
 
     private static final Logger LOG = LoggerFactory.getLogger(Security.class);
+       
+    private final static List<Integer> supportedResources = Arrays.asList(OSCORE_Master_Secret,
+            OSCORE_Sender_ID, OSCORE_Recipient_ID, OSCORE_AEAD_Algorithm, OSCORE_HMAC_Algorithm,
+            OSCORE_Master_Salt);
 
-    private byte[] masterSecret;
+    private String masterSecret;
     private String senderId;
     private String recipientId;
     private int aeadAlgorithm;
-    private int hmacAlgorithm;
+    private int hkdfAlgorithm;
+    private String masterSalt;
+    private String idContext;
 
     public Oscore() {
 
     }
 
-    public Oscore(byte[] masterSecret, String senderId, String recipientId, int aeadAlgorithm, int hmacAlgorithm) {
-        super();
-        this.masterSecret = masterSecret.clone();
+    /**
+     * Returns a new OSCORE instance.
+     */
+    public static Oscore set(String masterSecret, String senderId, String recipientId, int aeadAlgorithm, int hkdfAlgorithm, String masterSalt) {
+        return new Oscore(masterSecret, senderId, recipientId, aeadAlgorithm, hkdfAlgorithm, masterSalt);
+    }
+    
+    /**
+     * Returns a new OSCORE instance (less parameters and using default values).
+     */
+    public static Oscore set(String masterSecret, String senderId, String recipientId) {
+        int defaultAeadAlgorithm = 10; //AES_CCM_16_64_128
+        int defaultHmacAlgorithm = -10; //HKDF_HMAC_SHA_256
+        
+        return new Oscore(masterSecret, senderId, recipientId, defaultAeadAlgorithm, defaultHmacAlgorithm, "");
+    }
```

Comment:
What is a benefits of both static "constructor" above ?

Reply:
You can create several constructors with different number of parameter.

(static constructor is a good way to name a constructor but the con is that you can not reused when you extend the class)

Reply:
> You can create several constructors with different number of parameter.
> 
> (static constructor is a good way to name a constructor but the con is that you can not reused when you extend the class)

Okay, so it would be better to have 1 or several normal constructors, but no static constructors for the Oscore class?

Reply:
Yep, as it seems you don't really need to "name" your constructor, I think it's better to have several "normal" constructors.

Reply:
Removed the static "constructors" in commit dc37bd9c5a597cb41ca91238edbfe76665b72c76